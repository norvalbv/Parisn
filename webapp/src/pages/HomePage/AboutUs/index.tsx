import Radial from 'components/SVG/Design';
import React, { ReactElement } from 'react';
import PersonHoldingPhone from 'TempSVGImages/PersonHoldingPhone';

const AboutUs = (): ReactElement => {
  return (
    <div className="flex relative items-center w-1/2 mx-auto gap-8">
      <Radial />
      <div>
        <span className="mb-8 inline-block uppercase">Parisn.com</span>
        <p>
          Launched in 2021 allows you to choose the price you pay for your items in a reverse
          bidding strategy technique. Starting at £1,000 per item, over time the price of the item
          drops to £0.
        </p>
      </div>
      <PersonHoldingPhone />
    </div>
  );
};

export default AboutUs;
