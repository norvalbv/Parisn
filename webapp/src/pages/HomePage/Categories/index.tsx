import React, { ReactElement, useState } from 'react';
import StyledLink from 'components/StyledLink';
import Radial from 'components/SVG/Design';
// import { useCollections } from 'services/DataApiService';
import Loader from 'components/Loading';
import { collections } from '__mocks__/dataApiMock';
import PageIndicators from 'components/PageIndicators';

const Categories = (): ReactElement => {
  // const { data } = useCollections();
  const [page, setPage] = useState(0);
  const data = collections;

  if (!data) return <Loader />;

  const pageCount = Math.ceil(data.length / 3);

  // Create a shallow copy of a data that is limited to three categories per page.
  const processedData = data.slice(page * 3, page * 3 + 3);

  return (
    <section className="relative flex flex-col items-center px-4 md:px-0">
      <Radial colour="purple" className="hidden md:block" />
      <p className="uppercase text-center">Uncover Unparalleled Elegance</p>
      <h3 className="mb-6 md:mb-10 text-center">
        <span className="logo-text-medium">PARISN</span>&apos;s Exceptional Collections
      </h3>
      <div className="flex flex-col md:flex-row items-center gap-6 md:gap-10">
        {processedData.map((collection) => (
          <div
            className="relative h-[24rem] md:h-[31.875rem] w-full md:w-[19.625rem] rounded"
            key={collection.collections}
          >
            <img
              src={collection.image}
              alt={`${collection.collections} collection cover`}
              className="h-full w-full object-cover rounded brightness-[.65]"
            />
            <div className="absolute bottom-0 flex h-[6rem] md:h-[8.25rem] w-full flex-col items-center justify-center gap-4 md:gap-6 bg-gradient-to-b from-primary-dark/30 via-primary-dark/20 to-primary-dark/20">
              <h4 className="font-thin leading-[1.8125rem] text-center">{collection.collections}</h4>
              <StyledLink to={`/collections/${collection.collections}`} text="View All" />
            </div>
          </div>
        ))}
      </div>
      <PageIndicators pages={pageCount} currentPage={page} setCurrentPage={setPage} className="mt-6 md:mt-10" />
    </section>
  );
};

export default Categories;
