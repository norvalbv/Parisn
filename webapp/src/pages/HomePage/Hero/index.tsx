import React, { ReactElement } from 'react';
import Button from 'components/Button';
import Carousel from './Carousel';

const Hero = (): ReactElement => {
  return (
    <header className="flex h-screen justify-between pt-[10.125rem]">
      <section className="w-[23.8125rem]">
        <p className="text-[1.5rem]">
          Unlock the Essence of Urban Luxury with Refined Shopping Redefined by
        </p>
        <h1 className="mt-1 mb-3 tracking-[.42rem]">PARISN</h1>
        <p className="text-sm text-primary-neutral">
          Limited Editions, Coveted Brands, and the Thrill of Reverse Bidding
        </p>
        <Button text="Shop Now" className="mt-[3.75rem] w-72" roundedBorders="none" />
      </section>
      <Carousel />
    </header>
  );
};

export default Hero;
