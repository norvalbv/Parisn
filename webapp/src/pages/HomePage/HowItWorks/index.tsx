import React, { ReactElement } from 'react';
import StyledLink from 'components/StyledLink/intex';
import Radial from 'components/SVG/Design';
import Phone from '../../../TempSVGImages/phone';

const HowItWorks = (): ReactElement => {
  return (
    <div className="my-10 px-20 relative">
      <div className="flex w-4/5 gap-4 items-center">
        <Phone />
        <div>
          <span>1st Choose Your Item</span>
          <p className="mb-6">
            Handpick from a carefully curated collection of exclusive brands and limited edition
            products.
          </p>
          <StyledLink to="/" />
        </div>
      </div>
      <Radial />
      <div className="flex flex-row-reverse w-4/5 gap-4 items-center relative">
        <div className="text-[32px]">
          <div className="opacity-40 ml-14">
            <span className="mr-8 text-[44px] font-normal">£</span>1,000.00
          </div>
          <div className="bg-secondary-dark rounded-3xl border border-neutral my-10 py-6 pl-10 w-96">
            <span className="mr-8 text-[44px] font-normal">£</span>500.00
          </div>
          <div className="opacity-40 ml-14">
            <span className="mr-8 text-[44px] font-normal">£</span>200.00
          </div>
        </div>
        <div>
          <span>2nd Set Your Price</span>
          <p className="mb-6">
            Set your desired price point, empowering you to take control of your shopping
            experience.
          </p>
          <StyledLink to="/" />
        </div>
      </div>
      <div className="flex w-4/5 gap-4 items-center">
        <Phone />
        <div>
          <span>3rd Win Or Wait</span>
          <p className="mb-6">
            Secure your purchase at your chosen price or hold out for a potentially better deal as
            prices continue to fall.
          </p>
          <StyledLink to="/" />
        </div>
      </div>
    </div>
  );
};

export default HowItWorks;
