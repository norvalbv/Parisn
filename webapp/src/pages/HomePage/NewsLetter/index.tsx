import React, { ReactElement } from 'react';

const Newsletter = (): ReactElement => {
  return <div>Newsletter</div>;
};

export default Newsletter;
