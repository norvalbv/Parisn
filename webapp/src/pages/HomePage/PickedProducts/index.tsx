import React, { ReactElement, useRef, useState } from 'react';
import Button from 'components/Button';
import ClockIcon from 'components/SVG/ClockIcon';
import Badge from 'components/Badge';
import NavigationArrows from 'components/NavigationArrows';
import classNames from 'utils/classNames';
import { products } from '__mocks__/dataApiMock';
import Loading from 'components/Loading';
import { timeLeft } from 'utils/timeLeft';

const PickedProducts = (): ReactElement => {
  const data = products;

  if (!data) return <Loading />;

  const ref = useRef<HTMLDivElement>(null);
  const containerWidth = ref.current?.offsetWidth || 0;

  const [indexedImageInCenter, setIndexedImageInCenter] = useState(0);
  return (
    <section className="my-40">
      <div className="flex items-center justify-between">
        <div className="mb-10">
          <h3>Dive into Our Handpicked Selection</h3>
          <p className="uppercase">
            Featured Products with Unmatched Style and Exciting Bidding Opportunities
          </p>
        </div>
        <NavigationArrows
          leftArrow={{
            className: !indexedImageInCenter ? '' : 'cursor-pointer',
            fill: !indexedImageInCenter ? '#B0B0B0' : 'white',
            onClick: (): void => {
              if (!indexedImageInCenter) return;
              setIndexedImageInCenter((indexedImageInCenter) => indexedImageInCenter - 1);
            },
          }}
          rightArrow={{
            className: indexedImageInCenter === data.length - 3 ? '' : 'cursor-pointer',
            fill: indexedImageInCenter === data.length - 3 ? '#B0B0B0' : 'white',
            onClick: (): void => {
              if (indexedImageInCenter === data.length - 3) return;
              setIndexedImageInCenter((indexedImageInCenter) => indexedImageInCenter + 1);
            },
          }}
        />
      </div>
      <div
        className={classNames(
          'relative flex min-w-min items-center gap-[3.75rem] overflow-hidden transition-all duration-500'
        )}
        style={{
          left: `-${!indexedImageInCenter ? 0 : containerWidth * indexedImageInCenter + 60}px`,
        }}
      >
        {data.map((product, idx) => (
          <section
            className={classNames('relative h-[31.875rem] w-[19.625rem] rounded')}
            key={product.id}
            ref={ref}
          >
            <div className="absolute top-[0.6875rem] flex w-full items-center justify-between px-3">
              <div className="flex items-center gap-2">
                <ClockIcon />
                {/* <span className="text-xxs font-normal">1hr 14mins</span> */}
                <span className="text-xxs font-normal">
                  {timeLeft({ start: product.startTime, end: product.endTime })}
                </span>
              </div>
              <Badge type={idx % 2 ? 'limited' : 'new'} />
            </div>
            <img
              src={product.image}
              alt={product.id}
              className="h-[18.5rem] w-full rounded-t object-cover"
            />
            <div className="h-[13.375rem] rounded-b bg-gradient-to-br from-primary-light/[.03] via-primary-light/5 to-primary-light/10 px-3 py-4">
              <div className="flex items-center justify-between">
                <h4 className="uppercase">{product.title}</h4>
                <section className="w-min">
                  <span className="text-xs text-primary-neutral">Current&nbsp;price</span>
                  <span className="inline-block font-medium uppercase leading-[1.1875rem] tracking-[0.08rem]">
                    Â£&nbsp;{product.price}
                  </span>
                </section>
              </div>
              <p className="mb-6 mt-3 text-xs font-thin leading-[1.1875rem] text-white/60">
                {product.description.slice(0, 68)}&nbsp;
                {product.description.length > 68 && '...'}
              </p>
              <div className="flex gap-4">
                <Button
                  text="Make A Bid"
                  theme="ghost"
                  roundedBorders="md"
                  className="h-12 w-32 text-xs"
                  size="custom"
                  fontWeight="semibold"
                />
                <Button
                  text="Buy Now"
                  theme="light"
                  roundedBorders="md"
                  className="h-12 w-44 text-xs"
                  size="custom"
                  fontWeight="semibold"
                />
              </div>
            </div>
          </section>
        ))}
      </div>
    </section>
  );
};

export default PickedProducts;
