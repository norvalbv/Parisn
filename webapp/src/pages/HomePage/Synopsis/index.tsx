import React, { ReactElement } from 'react';

const Synopsis = (): ReactElement => {
  return (
    <div className="px-[25rem] text-center">
      <span className="inline-block">the Power of Reverse Bidding in 3 Simple Stepss</span>
      <span className="font-semibold inline-block text-[28px] leading-9">
        Master the Art of Smart Shopping at
        <span className="block font-semibold text-[42px]">PARISN</span>
      </span>
    </div>
  );
};

export default Synopsis;
