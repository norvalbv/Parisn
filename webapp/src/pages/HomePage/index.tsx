import { ReactElement } from 'react';
import HowItWorks from './HowItWorks';
import Categories from './Categories';
import Hero from './Hero';
import PickedProducts from './PickedProducts';
import Synopsis from './Synopsis';
import AboutUs from './AboutUs';
import Newsletter from './NewsLetter';

const HomePage = (): ReactElement => {
  return (
    <div>
      <Hero />
      <Synopsis />
      <HowItWorks />
      <Categories />
      <PickedProducts />
      <AboutUs />
      <Newsletter />
    </div>
  );
};

export default HomePage;
