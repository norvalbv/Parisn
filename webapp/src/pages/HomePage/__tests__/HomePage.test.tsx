import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import HomePage from '..';

describe('<HomePage />', () => {
  test('HomePage props with mandatory props', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <HomePage />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
