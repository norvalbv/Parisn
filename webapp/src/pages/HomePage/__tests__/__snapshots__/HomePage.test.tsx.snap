// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> HomePage props with mandatory props 1`] = `
[
  <div
    className="flex items-center overflow-hidden"
  >
    <div
      className="h-screen"
    >
      <video
        autoPlay={true}
        className="absolute h-[100vh] opacity-50 sepia contrast-125"
        loop={true}
        muted={true}
        src="https://cdn.shopify.com/videos/c/o/v/fe61f36873284fd2b4368f4a8fde0c9b.mp4"
      />
      <div
        className="absolute h-screen bg-fuchsia-200 opacity-10 w-screen"
      />
    </div>
    <div
      className="relative flex justify-center items-center flex-1"
    >
      <div
        className="absolute text-center"
      >
        <div
          data-testid="PARISN.com"
        >
          <span
            className="text-[3rem] drop-shadow-[0_0_16px_rgba(255,255,255,0.7)]"
          >
            DEMO SITE
          </span>
          <span
            className="drop-shadow-[0_0_16px_rgba(255,255,255,0.5)]"
          >
            .com
          </span>
        </div>
        <span
          className="text-3xl tracking-widest block"
        >
          A Fashion Company
        </span>
        <div
          className="mt-20 flex gap-10 items-center justify-center"
        >
          <button
            className="relative inline-flex items-center justify-center  uppercase  px-8 py-2.5 text-base w-full  none py-4 hover:scale-110 transition-all"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="button"
            style={
              {
                "backgroundColor": undefined,
                "color": undefined,
                "width": "15rem",
              }
            }
            type="button"
          >
            <span
              className="inline-block font-semibold"
            >
              View Collections
            </span>
          </button>
          <button
            className="relative inline-flex items-center justify-center  uppercase  px-8 py-2.5 text-base w-full  none py-4 hover:scale-110 transition-all"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="button"
            style={
              {
                "backgroundColor": undefined,
                "color": undefined,
                "width": "15rem",
              }
            }
            type="button"
          >
            <span
              className="inline-block font-semibold"
            >
              How It Works
            </span>
          </button>
          <button
            className="relative inline-flex items-center justify-center  uppercase  px-8 py-2.5 text-base w-full  none py-4 hover:scale-110 transition-all"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="button"
            style={
              {
                "backgroundColor": undefined,
                "color": undefined,
                "width": "15rem",
              }
            }
            type="button"
          >
            <span
              className="inline-block font-semibold"
            >
              Contact Us
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="h-[150vh] relative overflow-hidden"
  >
    <div
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <svg
        fill="#555555"
        height={0}
        stroke="#ffffff"
        width={0}
      >
        <line
          className="visx-line"
          fill="transparent"
          shapeRendering="crispEdges"
          stroke="#FAFAFA"
          strokeOpacity={0.75}
          x1={0}
          x2={0}
          y1={0}
          y2={0}
        />
        <line
          className="visx-line"
          fill="transparent"
          shapeRendering="crispEdges"
          stroke="#FAFAFA"
          strokeOpacity={0.75}
          x1={0}
          x2={0}
          y1={0}
          y2={0}
        />
        <line
          className="visx-line"
          fill="transparent"
          shapeRendering="crispEdges"
          stroke="#FAFAFA"
          strokeOpacity={0.75}
          x1={0}
          x2={0}
          y1={0}
          y2={0}
        />
        <line
          className="visx-line"
          fill="transparent"
          shapeRendering="crispEdges"
          stroke="#FAFAFA"
          strokeOpacity={0.75}
          x1={0}
          x2={0}
          y1={0}
          y2={0}
        />
      </svg>
      <p
        className="absolute text-primary-light text-center tracking-widest text-lg rotate-[8.85deg]"
        style={
          {
            "left": -50,
            "top": 0,
          }
        }
      >
        A new way of shopping
      </p>
      <p
        className="absolute text-primary-light text-center tracking-widest text-lg block rotate-[15deg]"
        style={
          {
            "left": -50,
            "top": 0,
          }
        }
      >
        Your way, your price
      </p>
    </div>
    <svg
      className="absolute rounded-full opacity-20 blur-3xl z-0 -top-32 -right-32"
      style={
        {
          "backgroundColor": "green",
          "height": "40rem",
          "width": "40rem",
        }
      }
    />
    <svg
      className="absolute rounded-full opacity-20 blur-3xl z-0 -bottom-32 -left-32"
      style={
        {
          "backgroundColor": "purple",
          "height": "40rem",
          "width": "40rem",
        }
      }
    />
  </div>,
  <div
    className="h-screen flex"
  >
    <p
      className="flex flex-col justify-center items-center text-center px-40 leading-8"
    >
      <span
        className="block font-semibold text-inherit underline"
      >
        PARSIN.com
      </span>
      Launched in 2022 allows you to choose the price you pay for your items in a reverse bidding strategy technique. Starting at £10,000 per item, over time the price of the item drops to £0.
    </p>
    <img
      alt="http://127.0.0.1:6001/images/dashboard-image.jpg"
      className="h-screen w-[42.5%]"
      src="http://127.0.0.1:6001/images/dashboard-image.jpg"
    />
  </div>,
]
`;
